test/e2e/encryption_at_rest_test.go

```diff
@@ -294,3 +302,223 @@ var _ = Describe("Encryption at rest AWS", Label("encryption-at-rest"), func() {
 		})
 	})
 })
+
+func configureManager(testData *model.TestDataProvider) {
+	mgr := actions.PrepareOperatorConfigurations(testData)
+	ctx := context.Background()
+	go func(ctx context.Context) context.Context {
+		err := mgr.Start(ctx)
+		Expect(err).NotTo(HaveOccurred())
+		return ctx
+	}(ctx)
+	testData.ManagerContext = ctx
+}
+
+func getProjectCondition(ctx context.Context, k8sClient client.Client, statusType status.ConditionType, ns string, name string) (*status.Condition, error) {
+	project := &v1.AtlasProject{}
+	err := k8sClient.Get(ctx, client.ObjectKey{Namespace: ns, Name: name}, project)
+	if err != nil {
+		return nil, err
+	}
+	for _, condition := range project.Status.Conditions {
+		if condition.Type == statusType {
+			return &condition, nil
+		}
+	}
+	return nil, fmt.Errorf("condition %s not found. found %v", statusType, project.Status.Conditions)
+}
+
+func createProjectWithValidationError(testData *model.TestDataProvider, errMsg string) {
+	if testData.Project.GetNamespace() == "" {
+		testData.Project.Namespace = testData.Resources.Namespace
+	}
+	By(fmt.Sprintf("Deploy Broken Project %s", testData.Project.GetName()), func() {
+		err := testData.K8SClient.Create(testData.Context, testData.Project)
+		Expect(err).ShouldNot(HaveOccurred(), "Project %s was not created", testData.Project.GetName())
+		Eventually(func(g Gomega) {
+			condition, err := getProjectCondition(
+				testData.Context,
+				testData.K8SClient,
+				status.ValidationSucceeded,
+				testData.Resources.Namespace,
+				testData.Project.GetName(),
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(condition).ToNot(BeNil())
+		}).WithPolling(5 * time.Second).WithTimeout(40 * time.Second).Should(Succeed())
```

Comment:
Can't we use this helper function instead of duplicating it?
https://github.com/mongodb/mongodb-atlas-kubernetes/blob/2592e1901e184503c8a1faa7f1d786b14743c7b1/test/e2e/actions/deploy/deploy_operator.go#L48

Reply:
`CreateProject` cannot replace `createProjectWithValidationError` because it:
- Will not look for the expected error, and end up as success if that is what happened.
- Will timeout trying to get a successful project created, which we do not expect in these test cases.

Reply:
I do feel that the code for this helper can be simplified somehow, though.